### PR TITLE
v6.1: add multi-level os support doc

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
v6.1: add multi-level os support doc

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/11134
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/br-deployment.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ This document describes the recommended deployment of Backup & Restore (BR) and
 
 Recommended practices when deploying BR:
 
-- In production environments, deploy BR on a node with at least 8 cores CPU and 16 GB memory. Select an appropriate OS version by following [Linux OS version requirements](/hardware-and-software-requirements.md#linux-os-version-requirements).
+- In production environments, deploy BR on a node with at least 8 cores CPU and 16 GB memory. Select an appropriate OS version by following [OS and platform version requirements](/hardware-and-software-requirements.md#os-and-platform-version-requirements).
```
Comment (TomShawn) [inline br/br-deployment.md:14]:
```suggestion
- In production environments, deploy BR on a node with at least 8 cores CPU and 16 GB memory. Select an appropriate OS version by following [OS and platform version requirements](/hardware-and-software-requirements.md#os-and-platform-requirements).
```

File: hardware-and-software-requirements.md [hunk 2]
```diff
@@ -7,7 +7,49 @@ summary: Learn the software and hardware recommendations for deploying and runni
 
 As an open source distributed NewSQL database with high performance, TiDB can be deployed in the Intel architecture server, ARM architecture server, and major virtualization environments and runs well. TiDB supports most of the major hardware networks and Linux operating systems.
 
-## Linux OS version requirements
+## OS and platform requirements
+
+<SimpleTab>
+<div label="For v6.1.1 and later v6.1.x versions">
+
+Since v6.1.1, TiDB provide multi-level supports of different quality standards for combinations of different operating systems and CPU architectures.
```
Comment (lilin90) [inline hardware-and-software-requirements.md:15]:
```suggestion
Starting from v6.1.1, TiDB provides multi-level support for different quality standards on the combination of operating systems and CPU architectures.
```

File: hardware-and-software-requirements.md [hunk 3]
```diff
@@ -7,7 +7,49 @@ summary: Learn the software and hardware recommendations for deploying and runni
...
+
++ For the following combinations of operating systems and CPU architectures, TiDB meets the requirements for enterprise-level production quality, and the product features have been comprehensively and systematically verified:
```
Comment (lilin90) [inline hardware-and-software-requirements.md:17]: How about this way? 
```suggestion
+ For the following combinations of operating systems and CPU architectures, TiDB **provides enterprise-level production quality**, and the product features have been comprehensively and systematically verified:
```

File: hardware-and-software-requirements.md [hunk 4]
```diff
@@ -7,7 +7,49 @@ summary: Learn the software and hardware recommendations for deploying and runni
...
+    | <ul><li>Red Hat Enterprise Linux 8.4 or a later 8.x version</li><li>CentOS 8.4 or a later 8.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>  |
+    | <ul><li>Red Hat Enterprise Linux 7.3 or a later 7.x version</li><li>CentOS 7.3 or a later 7.x version</li></ul>  |  <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+    | Kylin Euler V10 SP1/SP2   |   <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+    | UOS V20                 |   <ul><li>x86_64</li><li>ARM 64</li></ul>   |
+
+    > **Note:**
+    >
+    > According to [CentOS Linux EOL](https://www.centos.org/centos-linux-eol/), the upstream support for CentOS ended on December 31, 2021.
+
++ For the following combinations of operating systems and CPU architectures, you can compile, build, and deploy TiDB. In addition, you can also use the basic features of OLTP, OLAP, and the data tools. However, TiDB **does not guarantee enterprise-level production quality requirements**:
```
Comment (lilin90) [inline hardware-and-software-requirements.md:30]: "guarantee ... requirements" is a little bit weird to me. 🤔
```suggestion
+ For the following combinations of operating systems and CPU architectures, you can compile, build, and deploy TiDB. In addition, you can also use the basic features of OLTP, OLAP, and the data tools. However, TiDB **does not guarantee enterprise-level production quality**:
```

File: hardware-and-software-requirements.md [hunk 5]
```diff
@@ -7,7 +7,49 @@ summary: Learn the software and hardware recommendations for deploying and runni
...
+    |  Fedora 35 or later   |  x86_64           |
+    |  openSUSE Leap later than v15.3 (not including Tumbleweed) |  x86_64           |
+    |  SUSE Linux Enterprise Server 15  |  x86_64                        |
+
+    > **Note:**
+    >
+    > - For Oracle Enterprise Linux, TiDB supports the Red Hat Compatible Kernel (RHCK) and does not support the Unbreakable Enterprise Kernel provided by Oracle Enterprise Linux.
+    > - Support for Ubuntu 16.04 will be removed in future versions of TiDB. Upgrading to Ubuntu 18.04 or later is strongly recommended.
+
++ If you are using the 32-bit version of an operating system listed in the preceding two tables, TiDB **is not guaranteed** to be compilable, buildable or deployable on the 32-bit operating system and the corresponding CPU architecture, or TiDB does not actively adapt to the 32-bit operating system.
```
Comment (lilin90) [inline hardware-and-software-requirements.md:47]: Use a more common word.
```suggestion
+ If you are using the 32-bit version of an operating system listed in the two tables above, TiDB **is not guaranteed** to be compilable, buildable or deployable on the 32-bit operating system and the corresponding CPU architecture, or TiDB does not actively adapt to the 32-bit operating system.
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Pull request merged**